language/llama3-405b/consolidate_results.py

```diff
@@ -0,0 +1,137 @@
+import argparse
+import evaluate
+import glob
+import nltk
+import numpy as np
+import os
+import pandas as pd
+import pickle
+
+from pathlib import Path
+from transformers import LlamaTokenizerFast
+from tqdm import tqdm
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--dataset-path",
+        type=str,
+        default=None,
+        help="Path to .pkl generated by processorca.py",
+    )
+    parser.add_argument(
+        "--run-outputs",
+        type=str,
+        default="run_outputs",
+        help="Output dir generated by accuracy run.",
+    )
+    parser.add_argument(
+        "--model-dir",
+        type=str,
+        default=None,
+        help="Path to Llamav2 HuggingFace repo clone",
+    )
+    parser.add_argument(
+        "--output-pkl-path",
+        type=str,
+        default="full_output.pkl",
+        help="Path to dump output to",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def load_dataset(p: os.PathLike):
+    print(f"Loading from {p}...")
+    return pd.read_pickle(p)
+
+
+def load_run_outputs(p: os.PathLike):
+    g = glob.glob(str(Path(p) / "q*.pkl"))
+
+    by_query_idx = dict()
+    for pkl_file in g:
+        print(f"Loading from {pkl_file}...")
+        with open(pkl_file, "rb") as f:
+            d = pickle.load(f)
+        assert len(d["query_ids"]) == len(d["outputs"])
+
+        for i in range(len(d["query_ids"])):
+            qid = d["query_ids"][i]
+            assert qid not in by_query_idx
+            by_query_idx[qid] = d["outputs"][i]
+
+    return by_query_idx
+
+
+def main(args):
+    # Set up decode and evaluation objects
+    tokenizer = LlamaTokenizerFast.from_pretrained(args.model_dir)
```

Comment:
i may be wrong, but would it better to use AutoTokenizer here?

Reply:
I also suggest to use AutoTokenizer for robustness

Reply:
[consolidate_results.py](https://github.com/mlcommons/inference/pull/1915/files#diff-3b991886c168015084b278c78cdd4f663c38c5ea4e3937b61d53a2a47f5b44df) has been removed